insights/core/archives.py

```diff
@@ -62,12 +62,15 @@ def _tar_flag_for_content_type(self, content_type):
             raise InvalidContentType(content_type)
         return flag
 
-    def from_path(self, path, extract_dir=None, content_type=None):
+    def from_path(self, path, extract_dir=None, content_type=None, decompressor=None):
```

Comment:
This is exposed through the extract function below, and extract can't give zipped files custom decompressors.

Reply:
I removed the decompressor option since we can bypass mimetype detection by passing content_type explicitly.